src/main/java/com/search/DepthFirstSearch.java

```diff
@@ -0,0 +1,126 @@
+package src.main.java.com.search;
+
+/**
+ * Searching is faster in sorted structures. Binary search is O(log n).
+ * However, the cost of sorting is O(n · log n).
+ * What to do when adding or removing elements? Sort again? No.
+ * Create efficient data structures to maintain sorted sequences, and search in them
+ * Key example: binary sorted tree, allowing O(log N) insert, remove and lookup.
+   
+   In comes, depth-first search
+ * Worst-case performance	O(n)
+ * Best-case performance	O(1)
+ * Average performance	   	O(n)
+ * 
+ * @author abir (https://github.com/abircb)
```

Comment:
Please do not add a signature inside the code. See [Contribution Guidelines](https://github.com/TheAlgorithms/Java/tree/Development#contribution-guidelines) for details.

Reply:
```suggestion

```